src/neuroconv/tools/nwb_helpers/_configure_backend.py

```diff
@@ -11,6 +11,86 @@
 from ..importing import is_package_installed
 
 
+def _resolve_nwbfile_object_ids(
+    nwbfile: NWBFile, backend_configuration: Union[HDF5BackendConfiguration, ZarrBackendConfiguration]
+) -> Dict[str, str]:
+    """
+    When a configuration is generated on an equivalent NWBFile elsewhere in memory, the object IDs will not match.
+
+    But a mapping can be formed to reconcile the two.
+    This is the case for the NWB GUIDE (since they are separate endpoints in the Flask server).
+
+    The default behavior forms a simple identity mapping.
+
+    Parameters
+    ----------
+    nwbfile : pynwb.NWBFile
+        The in-memory pynwb.NWBFile object to configure.
+    backend_configuration : HDF5BackendConfiguration or ZarrBackendConfiguration
+        The configuration model to use when configuring the datasets for this backend.
+
+    Returns
+    -------
+    dataset_configuration_to_nwbfile_object_ids: Dict[str, str]
+        A mapping of dataset configuration object IDs to their equivalent NWBFile object IDs.
+    """
+    nwbfile_objects = nwbfile.objects
+
+    nwbfile_object_ids = [nwbfile_object_id for nwbfile_object_id in nwbfile_objects.keys()]
+    dataset_configuration_object_ids = [
+        dataset_configuration.object_id in nwbfile_object_ids
+        for dataset_configuration in backend_configuration.dataset_configurations.values()
+    ]
+
+    # Default to identity
+    dataset_configuration_to_nwbfile_object_ids = {
+        dataset_configuration_object_id: dataset_configuration_object_id
+        for dataset_configuration_object_id in dataset_configuration_object_ids
+    }
+    if not all(
```

Comment:
I prefer to use specific named variables for this long boolean conditions. Can we call the thing in parenthesis something like `all_ids_in_dataset_configuration_are_in_nwbfile`?

Reply:
Done